Simbody/src/GeneralForceSubsystem.cpp

```diff
@@ -34,12 +34,457 @@
 #include "simbody/internal/GeneralForceSubsystem.h"
 #include "simbody/internal/SimbodyMatterSubsystem.h"
 #include "simbody/internal/MultibodySystem.h"
+#include <iostream>
+#include <exception>
 
 #include "ForceImpl.h"
 
+#include <memory>
+
+//Threading constants used by CalcForcesTask
+namespace {
+const int NumNonParallelThreads = 1;
+const int NonParallelForcesIndex = 0;
+}
 
 namespace SimTK {
 
+/** Calculates each enabled force's contribution in the MultibodySystem.
+CalcForcesTask allows force calculations to occur in parallel with non-parallel
+forces being calculated on Thread NonParallelForcesIndex and all other parallel
+forces being calculated on seperate threads.*/
+class CalcForcesTask : public ParallelExecutor::Task {
```

Comment:
This is intended as a file-scope "helper" class, but has a couple of problems:
- It is defined in the SimTK namespace and is thus globally visible, and
- It has a doxygen comment but we don't let doxygen process .cpp files.

File-scope class definitions should be in the anonymous namespace like you have the constants in above, so that they can't conflict with coincidentally-named classes elsewhere.